client/src/GamemasterInGame/GameMasterActions.tsx

```diff
@@ -21,6 +21,9 @@ export function GameMasterActions({ selectedRoles }: GameMasterActionsProps) {
   const { game } = useDefiniteGame();
   const [distributeRolesError, , , distributeRoles, clear] =
     useDistributeRoles();
+  const allPlayersSeenRoles =
+    Object.keys(game.playerHasSeenRole).length === game.playerList.length &&
+    Object.values(game.playerHasSeenRole).every(Boolean);
```

Comment:
Alright so there's no actual reason for playerHasSeenRoles to be a Record<player, boolean>